### PR TITLE
Add stream method

### LINKED ISSUE #1: Add .streaming method to Limiter
The `Limiter` should expose a convenience method which returns a `Stream[Limiter]`. The `bracket` pattern should be mentioned in the README, accordingly.

### PR DESCRIPTION
**Pull request by keirlawson**
Resolves #1

### CODE REVIEW

**Comment by SystemFw** [comment]
Nice, thanks!
Could you slightly reword the README to actually say that `.stream` is the recommended option? 
For example because it's guaranteed to cleanup things even on errors

**Comment by keirlawson** [comment]
Have switched the example given in the README to use `StreamApp` and `Limiter.stream`

File: README.md [hunk 1]
```diff
@@ -92,29 +93,19 @@ you can then do:
...
-  val managed = for {
-    limiter <- Limiter.start[IO](100 every 1.minute)
-    res <- YourWholeProgram(limiter.worker).doStuff
-    _ <- client.shutdown
-  } yield res
+object Main extends StreamApp {
+  def stream(args: List[String], requestShutdown: IO[Unit]) : Stream[IO, ExitCode] =
+    for {
+      limiter <- Limiter.stream[IO](100 every 1.minute)
+      res <- YourWholeProgram(limiter.worker).doStuff
```
Comment (SystemFw) [inline README.md]: Missing a `Stream.eval` here, currently won't compile

Comment (keirlawson) [reply README.md]: Oops!

**Comment by keirlawson** [comment]
Ok that should be the example working now, should really set up tut to catch this stuff!

**Comment by SystemFw** [comment]
Thanks a lot :)

**Pull request merged**